### PR TITLE
ci: comparative benchmarks for vm, core, and trie operations

### PR DESCRIPTION
**Pull request by meowsbits**
Rel #283

Provides 
- new benchmarking functionality for the JSON VM tests suite (`tests/vm_bench_test.go`)
- 3 new Github Actions workflows that compare go-ethereum vs. core-geth benchmarks for
  + the JSON VM tests,
  + `core/`, and
  + `trie/` packages.

Intention is to provide a reliable means of establishing efficiency and optimization comparison with upstream source.
Example test runs can be seen on my personal fork at https://github.com/meowsbits/core-geth/actions. 

The actions' workflows run go-ethereum and core-geth jobs in parallel, and compare the results using `benchstat`.

### CODE REVIEW

File: .github/workflows/bench-core.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        id: go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ^1.16
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+        with:
+          repository: 'ethereum/go-ethereum'
+          ref: 'v1.9.25'
```
Comment (ziogaschr) [inline .github/workflows/bench-core.yml:47]: It's important to not forget to update the version here from now on. Wondering if we can fetch this version from somewhere in another step and use it here.

**Review by ziogaschr — APPROVED** [review]
Nice work man. It's useful to be able to keep an eye on such stats.

File: .github/workflows/bench-core.yml (hunk 1 above)
Comment (meowsbits) [reply .github/workflows/bench-core.yml:47]: Yea, this crossed my mind too. The issue with dynamic updates would be that there are no guarantees that benchmark tests will stay constant across upstream versions (although in practice, this is probably an edge case); and I wanted to maintain consistency for the test control group. My thinking was that when we start to actually use these, we _should_ expect ourselves to have the basic states of the test groups in mind when analyzing their results.

I'm not hard and fast on hardcoding it, but that was my reasoning. If you'd like to push for dynamic versioning I would be more than happy to see what you have in mind.

**Comment by meowsbits** [comment]
Just rebased on master@f0dd598bf86747d87ca9691fe668631cc18ae138.
